modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -729,29 +740,36 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
             ncn = std::min(ncn, inpCn);
             p.blk_size_cn = ncn;
 
-            int dil_d = !isConv2D? dilations[0] : 1;
-            int dil_h = dilations[dilations.size() - 2];
+            int dil_d = isConv3D? dilations[0] : 1;
+            int dil_h = isConv1D? 1 : dilations[dilations.size() - 2];
             int dil_w = dilations.back();
 
             p.ofstab_.resize(karea * ncn);
             int* ofstab = &p.ofstab_[0];
 
-            if (isConv2D)
+            if (isConv1D)
             {
                 for( int k = 0; k < ncn; k++ )
-                    for( int k_r = 0; k_r < kernel_h; k_r++ )
-                        for( int k_c = 0; k_c < kernel_w; k_c++ )
-                            ofstab[(k*kernel_h + k_r)*kernel_w + k_c] =
-                                   (k*height + k_r*dil_h)*width + k_c*dil_w;
+                    for( int k_c = 0; k_c < kernel_w; k_c++ )
+                        ofstab[k*kernel_w + k_c] = k*width + k_c*dil_w;
+            }
+            else if (isConv2D)
+            {
+                for (int k = 0; k < ncn; k++)
+                    for (int k_r = 0; k_r < kernel_h; k_r++)
+                        for (int k_c = 0; k_c < kernel_w; k_c++)
+                            ofstab[(k * kernel_h + k_r) * kernel_w + k_c] =
+                                    (k * height + k_r * dil_h) * width + k_c * dil_w;
```

Comment:
Can you please keep origin formatting so this part of code can be unchanged? Thanks!